ovssubnet/registry/registry.go

```diff
@@ -42,24 +42,14 @@ func newNodeEvent(action, key, value string) *api.NodeEvent {
 	if key != "" {
 		_, nodeEvent.Node = path.Split(key)
 
-		var node map[string]interface{}
-		err := json.Unmarshal([]byte(value), &node)
+		nodeIP, err := getNodeIP(value)
 		if err == nil {
-			nodeStatus, ok := node["Status"].(map[string]interface{})
-			if ok {
-				nodeAddresses, ok := nodeStatus["Addresses"].([]interface{})
-				if ok {
-					nodeAddressMap, ok := nodeAddresses[0].(map[string]interface{})
-					if ok {
-						nodeEvent.NodeIP = nodeAddressMap["Address"].(string)
-						return nodeEvent
-					}
-				}
-			}
+			nodeEvent.NodeIP = nodeIP
+			return nodeEvent
 		}
 	}
 
-	fmt.Printf("Error decoding node event: nil key (%s,%s,%s).\n", action, key, value)
+	fmt.Errorf("Error decoding node event: nil key (%s,%s,%s).\n", action, key, value)
```

Comment:
There is a lot of inconsistency in registry.go for reporting errors. I'll handle this in a separate PR.